gossip/src/cluster_info.rs

```diff
@@ -1398,22 +1393,28 @@ impl ClusterInfo {
         } else {
             peers.iter().map(|peer| &peer.tvu).collect()
         };
-        let mut sent = 0;
-        while sent < dests.len() {
-            match multicast(s, &packet.data[..packet.meta.size], &dests[sent..]) {
-                Ok(n) => sent += n,
-                Err(e) => {
-                    inc_new_counter_error!(
-                        "cluster_info-retransmit-send_to_error",
-                        dests.len() - sent,
-                        1
-                    );
-                    error!("retransmit result {:?}", e);
-                    return Err(GossipError::Io(e));
+        let mut dests = &dests[..];
+        let data = &packet.data[..packet.meta.size];
+        while !dests.is_empty() {
+            match multicast(s, data, dests) {
+                Ok(n) => dests = &dests[n..],
+                Err(err) => {
+                    inc_new_counter_error!("cluster_info-retransmit-send_to_error", dests.len(), 1);
```

Comment:
nit: maybe this error should be called `cluster_info-retransmit-multicast_error`, and the `cluster_info-retransmit-error` error below should be called `"cluster_info-retransmit-send_to_error"?

Reply:
I agree that the error names are misleading, but I intentionally didn't want to change them so that in the period of time the cluster is upgrading to the new code we can track the error rates here and compare it with before.
For example, if half of the cluster upgrades to this new code, and the other half doesn't, then changes in the metric would be very confusing if the metric name has changed as well.

I would rather update the metrics name once the cluster has already updated to this new code and the code seems to be working fine.